tests/replicaset/manager-getservers-001.phpt

```diff
@@ -76,7 +76,7 @@ array(3) {
       ["primary"]=>
       string(19) "192.168.112.10:3000"
       ["tags"]=>
-      object(stdClass)#%d (%d) {
+      array(2) {
```

Comment:
Well, without PHPC-315 and introducing a new option to control the top-level document, we have no way to request native array decoding with `bson_to_zval()` when we want it (e.g. debug output). Otherwise, we'd always get a stdClass back unless the ODM behavior picks up a class name in our binary field.

I think PHPC-315 should be integrated into this PR, and then I can replace the `state.map.document_type` assignments with `state.map.root_type`.